docs/examples/go-ipfs-as-a-library/main.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	config "github.com/ipfs/go-ipfs-config"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	libp2p "github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/plugin/loader" // This package is needed so that all the preloaded plugins are loaded automatically
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	iCore "github.com/ipfs/interface-go-ipfs-core"
+	iCorePath "github.com/ipfs/interface-go-ipfs-core/path"
```

Comment:
nit: package names don't usually use caps

Reply:
Good to know. It is a usually as in "us, the go-ipfs team" or "us, the golang community"? Tunning up my codestyle to match what is convention :)

Reply:
Fixed :)

Reply:
https://blog.golang.org/package-names#TOC_2.

Reply:
Good to know: https://github.com/golang/go/wiki/CodeReviewComments 

(but ignore https://github.com/golang/go/wiki/CodeReviewComments#imports)

---

Also something that's good to know https://github.com/golang/go/wiki/SliceTricks

---

And in case you didn't know, https://www.evanjones.ca/go-gotcha-loop-variables.html.

---

![image](https://user-images.githubusercontent.com/310393/67747724-976fed80-f9e6-11e9-91f4-ead9ea94a9ee.png)